### DISCUSSION TITLE
what software are inside GitHub enterprise server?

### DISCUSSION THREAD

**Discussion by anlexN** [community/community#84958] [Enterprise]
### Select Topic Area

Question

### Body

I do not find these informations in the GitHub docs. I want to try GitHub enterprise server, but first I need to know your tech stack.

**Comment by PhilipPanda** [comment]
To learn about the tech stack of GitHub Enterprise Server, it's best to contact GitHub's support or sales team.

**Reply by anlexN** [reply]
Hello, Have you tried GitHub Enterprise Server?

**Comment by anlexN** [comment]
@github/support, please help me.

**Reply by anlexN** [reply]
@PhilipPanda @mgriffin @zackfern @elisealix22 @dav1dc-github @grandmas-favorite @gitstua @ghostinhershell @rileybroughten

**Comment by emad313** [comment]
Here is documentation or discussion about the GitHub Enterprise server.

https://docs.github.com/en/enterprise-server@3.10/admin/overview/about-github-enterprise-server

**Reply by anlexN** [reply]
I have reviewed all docs in the [Enterprise administrators](https://docs.github.com/en/enterprise-cloud@latest/admin), I can guess some software, but still be confused to me.
No answer to me, so sad.

**Comment by mishmanners** [comment]
GitHub Enterprise Server covers a lot of different features, such as:
- Code scanning
- Private repositories
- GitHub Pages
- Codespaces
- GitHub CLI 
- Manage your organisation, rules, PRs, etc
- GitHub Actions
- GitHub Advanced Security 
- GitHub Packages
- and more

It's designed to help you build, secure, and ship software faster than ever before.

Each instance of GitHub Enterprise Server is tailored specifically to you and your company's needs. To try it out and see what you and your team might like to use, and which features you'd like to have access to, contact the GitHub Sales team: https://github.com/enterprise/contact

**Reply by anlexN** [reply]
I have sent two emails to sales team for almost 15 days, but nobody reply me. so sad! can you help me follow?

**Reply by mishmanners** [reply]
Did you email or use the form?

**Reply by anlexN** [reply]
form. what happened?

**Comment by anlexN** [comment]
hello, who can help me?

**Comment by mgriffin** [comment]
GitHub Enterprise Server is provided as an appliance image, and any modifications to the appliance are not supported. Our [system overview](https://docs.github.com/en/enterprise-server@3.11/admin/overview/system-overview) page has some information about the stack. It's based on a custom Ubuntu image which gets updated by the GitHub Enterprise Server update images.

You can find out the list of [open source dependencies](https://docs.github.com/en/enterprise-server@3.11/admin/overview/system-overview#open-source-dependencies-for-github-enterprise-server) either from the appliance itself, or from the "metadata" tab on the download page. The link is in the last link there.

GitHub Enterprise Server runs as a virtual appliance, either on your own hypervisor hardware or in the cloud. Once it's installed, you don't need to do any work with the anything in the stack, it's all self contained.

**Reply by anlexN** [reply]
@mgriffin 

I have downloaded [dependencies-3.11.4-kvm.tar.gz](https://github-enterprise.s3.amazonaws.com/metadata/3.11.4/dependencies-3.11.4-kvm.tar.gz)

how to open dependencies-3.11.4-kvm.tar.gz correctly? have some tricks?

**Reply by mgriffin** [reply]
That file is compressed, so you need to uncompress it before you can see the contents. The quickest way is to use the [`tar` command](https://www.man7.org/linux/man-pages/man1/tar.1.html) like this:

```bash
tar zxvf dependencies-3.11.4-kvm.tar.gz
```

`tar` is usually installed on a Linux instance, but if you're using a different operating system, you may have to uncompress the file in a different way.

**Comment by Lanrad23** [comment]
As I recall, it runs on Ruby on Rails, a popular web application framework. Basically, Ruby on Rails is responsible for handling most of the application logic.
